util/git/client.go

```diff
@@ -57,15 +60,33 @@ func NewFactory() ClientFactory {
 	return &factory{}
 }
 
-func (f *factory) NewClient(rawRepoURL, path, username, password, sshPrivateKey string, insecureIgnoreHostKey bool) (Client, error) {
+func (f *factory) NewClient(rawRepoURL, path, username, password, sshPrivateKey string, insecure bool) (Client, error) {
 	var creds Creds
 	if sshPrivateKey != "" {
-		creds = SSHCreds{sshPrivateKey, insecureIgnoreHostKey}
+		creds = SSHCreds{sshPrivateKey, insecure}
 	} else if username != "" || password != "" {
 		creds = HTTPSCreds{username, password}
 	} else {
 		creds = NopCreds{}
 	}
+
+	// We need a custom HTTP client for go-git when we want to skip validation
+	// of the server's TLS certificate (--insecure-ignore-server-cert). Since
+	// this change is permanent to go-git Client during runtime, we need to
+	// explicitly replace it with default client for repositories without the
+	// insecure flag set.
+	if IsHTTPSURL(rawRepoURL) {
+		if insecure {
```

Comment:
can we always use this library? it's got to work, so lets just always use it

Reply:
Sure, it's basically the counterpart to IsSSHURL() - but as of know, just a wrapper around a regexp's MatchString(), and not thoroughly tested yet.

Reply:
I don't think I was clear, I'd like to avoid the if/else here on line 79. I think you can always use the `customHttpClient`?